servers/src/main/java/tachyon/worker/block/BlockMasterSync.java

```diff
@@ -153,6 +155,13 @@ public void run() {
               + cmdFromMaster.toString(), e);
         }
         mMasterClient.resetConnection();
+        try {
+          setWorkerId();
+          registerWithMaster();
```

Comment:
My question about the BlockMasterSync is that if at any point, registerWithMaster() fails, should the BlockMasterSync just ignore the exception, and try to get a new id and try again? I think the current code, and your new modifications will just crash the BlockMasterSync whenever registerWithMaster() fails.

One case where registerWithMaster() may fail is when this thread already got an id, but the master failed over before being able to call register(). I don't think that should crash the sync thread. What do you think about this?